### ISSUE TITLE
homu can't see successful Travis run

### ISSUE THREAD

**Issue by Valloric** [servo/homu#116]
We (YCM & ycmd devs) used to use the homu.io service, but that is now mostly defunct so we're trying to switch to a custom instance.

I'm running the latest master on GCE here: http://35.185.255.222:54856

[This is the pull request](https://github.com/Valloric/ycmd/pull/754#issuecomment-300977426) I'm using to test out this integration, and here is [the successful run on Travis](https://travis-ci.org/Valloric/ycmd/builds/231784868) that was kicked off [by this comment](https://github.com/Valloric/ycmd/pull/754#issuecomment-301221997) (zzbot is our homu bot user).

No matter what I do, I can't get homu to notice that the travis run succeeded. The [ycmd queue](http://35.185.255.222:54856/queue/ycmd) is currently showing PR 754 as "pending", though that will time out soon and switch to "failed".

From my understanding, there's 2 ways that homu can learn about successful builds: through the "status" event from the Github webhook and from the Travis webhook. The Travis webhook as implemented by homu doesn't work anymore since [Travis changed the auth mechanism](https://blog.travis-ci.com/2016-08-31-webhook-delivery-changes) back in November 2016 and the code hasn't been updated to support the new method. This is discussed in [your issue 60](https://github.com/servo/homu/issues/60).

So unless I'm missing something, only the Github status events should notify homu of completion, but I don't know what I'm doing wrong here that's causing homu to fail to see success.

I'm including my `cfg.toml` below (with secrets replaced by "xxx") to help debugging:

```
[github]

# Information for securely interacting with GitHub. These are found/generated
# under <https://github.com/settings/applications>.

# A GitHub personal access token
access_token = "xxx"

# A GitHub oauth application for this instance of homu:
app_client_id = "xxx"
app_client_secret = "xxx"

[git]

# Use the local Git command. Required to use some advanced features. It also
# speeds up Travis by reducing temporary commits
#local_git = false

# SSH private key. Needed only when the local Git command is used
#ssh_key = """
#"""

# By default, Homu extracts the name+email from the Github account.  However,
# you may want to use a private email for the account, and associate the commits
# with a public email address.
#user = "Some Cool Project Bot"
#email = "coolprojectbot-devel@example.com"

[web]

# The port homu listens on
port = 54856

# Synchronize all open PRs on startup
sync_on_start = true

# Custom hooks can be added as well
# Homu will ping the given endpoint with POSTdata of the form:
# {'body': 'comment body', 'extra_data': 'extra data', 'pull': pull req number}
# The extra data is the text specified in `@homu hookname=text`
#
# [hooks.hookname]
# trigger = "hookname" # will be triggered by @homu hookname or @homu hookname=text
# endpoint = "http://path/to/endpoint"
# access = "try" # access level required
# has_response = true # Should the response be posted back to github? Only allowed if realtime=true
# realtime = true # Should it only run in realtime mode? If false, this will be replayed each time homu is started (probably not what you want)

# An example configuration for repository (there can be many of these)
[repo.ycmd]

# github.com/<owner>/<name>
owner = "Valloric"
name = "ycmd"

# who has r+ rights?
reviewers = []
# Alternatively, set this to allow any github collaborator;
# note that you can *also* specify reviewers above.
auth_collaborators = true

# who has 'try' rights? (try, retry, force, clean, prioritization)
try_users = []

# Keep the commit history linear. Requires the local Git command
#linear = false

# Auto-squash commits. Requires the local Git command
#autosquash = true

# If the PR already has the same success statuses that we expect on the auto
# branch, then push directly to branch if safe to do so. Requires the local Git
# command.
#status_based_exemption = false

## branch names (these settings here are the defaults)
#[repo.NAME.branch]
#
#auto = "auto"
#try = "try"
#rollup = "rollup"

[repo.ycmd.github]

# arbitrary secret (e.g. openssl rand -hex 20)
secret = "xxx"

## Use buildbot for running tests
#[repo.NAME.buildbot]
#
#url = ""
#secret = ""
#
#builders = ["auto-linux", "auto-mac"]
#try_builders = ["try-linux", "try-mac"]
#
#username = ""
#password = ""

## Use travis for running tests
[repo.ycmd.travis]
## found under <https://travis-ci.org/profile/info>.
token = "xxx"

[repo.ycmd.status.travis]
context = 'continuous-integration/travis-ci/push'

[repo.ycmd.status.appveyor]
context = 'continuous-integration/appveyor/branch'

## Use the Status API
#[repo.NAME.status.LABEL]
#
## String label set by status updates
#context = ""
#
## Equivalent context to look for on the PR itself if checking whether the
## build should be exempted. If omitted, looks for the same context. This is
## only used if status_based_exemption is true.
#pr_context = ""

[repo.YouCompleteMe]
owner = "Valloric"
name = "YouCompleteMe"
reviewers = []
auth_collaborators = true
try_users = []

[repo.YouCompleteMe.github]
secret = "xxx"

[repo.YouCompleteMe.status.travis]
context = 'continuous-integration/travis-ci/push'

[repo.YouCompleteMe.status.appveyor]
context = 'continuous-integration/appveyor/branch'

[repo.YouCompleteMe.travis]
token = "xxx"

## The database homu uses
[db]

## SQLite file
file = "main.db"
```

**Comment by Manishearth** [comment]
Make sure the status role is enabled in your homu webhook settings on GitHub.

**Comment by Manishearth** [comment]
Also, you'll have to remove the `[repo.ycmd.travis]` if you're not using it since homu will gate on all CI you tell it to.

**Comment by Valloric** [comment]
Thanks for the quick response!

> Make sure the status role is enabled in your homu webhook settings on GitHub.

It is indeed enabled.

> Also, you'll have to remove the [repo.ycmd.travis] if you're not using it since homu will gate on all CI you tell it to.

Not sure I'm following this. I understand you're telling me to remove the travis secret token, but I'm not sure I get the "since homu will gate on all CI you tell it to" part. Are you saying that if I have that token present, homu will insist on getting an event on the `/travis` endpoint?

**Comment by Valloric** [comment]
@Manishearth Another thing I noticed that could be related to this: when homu kicks off a build on travis, it doesn't kick off one on appveyor. You can see stuff is happening on the "auto" branch on Travis, but silence on appveyor.

https://ci.appveyor.com/project/Valloric/ycmd/history
https://travis-ci.org/Valloric/ycmd/branches

**Comment by Manishearth** [comment]
Well yes, you have to tell appveyor to build on the auto branch.

Homu doesn't know anything about kicking off builds or other CI-specific things, it only knows how to listen for status hooks.

> I understand you're telling me to remove the travis secret token, but I'm not sure I get the "since homu will gate on all CI you tell it to" part. Are you saying that if I have that token present, homu will insist on getting an event on the /travis endpoint?

 Not just the token, that entire section of the config. You have given homu three CI options, it will wait for all three to complete.

**Comment by Valloric** [comment]
> Well yes, you have to tell appveyor to build on the auto branch.

I was just about to tell you how I have indeed configured Appveyor to build on all branches, but before I did that, I wanted to triple check. And yes, we do have "all branches" set for the "branches to build" option in Appveyor UI... [but we also have this.](https://github.com/Valloric/ycmd/blob/master/appveyor.yml#L3)

So that explains that. I'll remove it and try again. :)

> Not just the token, that entire section of the config. You have given homu three CI options, it will wait for all three to complete.

Ah, makes sense. Thanks for clarifying! You've been extremely helpful!
